sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -216,6 +242,21 @@ def plot(self, plot_method="contourf", ax=None, xlabel=None, ylabel=None, **kwar
         if ax is None:
             _, ax = plt.subplots()
 
+        estimator_name_to_display = (
+            estimator_name if estimator_name is not None else self.estimator_name_
+        )
+        response_method_to_display = (
+            response_method_used
+            if response_method_used is not None
+            else self.response_method_used_
+        )
+
+        if estimator_name_to_display or response_method_to_display:
+            ax.set_title(
+                f"Boundary: {estimator_name_to_display} "
+                f"Method: {response_method_to_display}"
+            )
```

Comment:
Not sure about this logic. Do we want to add a title if only `response_method_to_display` is provided?
If we did, we'd have to change the title to e.g., 'Method: predict', or '"predict" method'